elisa/src/org/labkey/elisa/ElisaSampleFilePropertyHelper.java

```diff
@@ -0,0 +1,119 @@
+import org.labkey.api.exp.property.DomainProperty;
+import org.labkey.api.reader.DataLoader;
+import org.labkey.api.reader.DataLoaderFactory;
+import org.labkey.api.reader.DataLoaderService;
+import org.labkey.api.study.assay.SampleMetadataInputFormat;
+
+import javax.servlet.http.HttpServletRequest;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ElisaSampleFilePropertyHelper extends PlateSampleFilePropertyHelper
+{
+    public ElisaSampleFilePropertyHelper(Container container, ExpProtocol protocol, List<? extends DomainProperty> domainProperties, PlateTemplate template, SampleMetadataInputFormat inputFormat)
+    {
+        super(container, protocol, domainProperties, template, inputFormat);
+    }
+
+    @Override
+    public Map<String, Map<DomainProperty, String>> getSampleProperties(HttpServletRequest request) throws ExperimentException
+    {
+        if (_sampleProperties != null)
+            return _sampleProperties;
+
+        File metadataFile = getSampleMetadata(request);
+        if (metadataFile == null)
+            throw new ExperimentException("No metadata or data file provided");
+
+        Map<String, Map<DomainProperty, String>> allProperties = new HashMap<>();
+        try
+        {
+            DataLoaderFactory factory = DataLoaderService.get().findFactory(metadataFile, null);
+            DataLoader loader = factory.createLoader(metadataFile, true);
+
+            String plateColumnName = "Plate Name";
+            String sampleColumnName = "Sample";
+            String wellLocationColumnName = "Well";
+            String spotColumnName = "Spot";
+
+            boolean hasPlateColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(plateColumnName));
+            boolean hasSampleColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(sampleColumnName));
+            boolean hasWellLocationColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(wellLocationColumnName));
+            boolean hasSpotColumn = Arrays.stream(loader.getColumns()).anyMatch(c -> c.getColumnName().equalsIgnoreCase(spotColumnName));
+
+            if (!hasPlateColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + plateColumnName + "\".");
+            if (!hasSampleColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + sampleColumnName + "\".");
+            if (!hasWellLocationColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + wellLocationColumnName + "\".");
+            if (!hasSpotColumn)
+                throw new ExperimentException("Sample metadata file does not contain required column \"" + spotColumnName + "\".");
```

Comment:
This code to stream the columns to find a given column name and then throw if not found is replicated here and in `HighThroughputImportHelper`. Is there anywhere that would make sense to factor this out to shared code?